src/Microsoft.Experimental.Collections/Microsoft/Collections/Extensions/OrderedDictionary.cs

```diff
@@ -109,13 +116,13 @@ public OrderedDictionary(IEnumerable<KeyValuePair<TKey, TValue>> collection, IEq
 
         public void Clear()
         {
-            if (Count > 0)
+            int count = Count;
+            if (count > 0)
             {
                 Array.Clear(_buckets, 0, _buckets.Length);
-                Array.Clear(_entries, 0, Count);
+                Array.Clear(_entries, 0, count);
                 Count = 0;
             }
-            ++_version;
```

Comment:
Depends what invalid means, if it means "will fail or produce incorrect results" such as in the case where an insert moves an element along in the array, causing the enumerator to potentially return an element more than once or not at all. In the case of Clear, the MoveNext() checks Count every time `if (_index < _orderedDictionary.Count)` so MoveNext() will return false as if it reached the end, which is arguably correct. If Current is already set, it will not be cleared, but we don't check version in that property getter anyway.

It's probably a bit pedantic, but it seems across our collections we never had a solid definition. I think generally we are trying to relax it a bit (eg we now allow Remove during enumeration on Dictionary)

My care level is low though so either is fine.

Reply:
Fair enough.